src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/send_request_to_opensearch.ts

```diff
@@ -94,79 +96,95 @@ export function sendRequestToOpenSearch(
       } // append a new line for bulk requests.
 
       const startTime = Date.now();
-      opensearch
-        .send(opensearchMethod, opensearchPath, opensearchData)
-        .always((dataOrjqXHR: any, textStatus: string, jqXhrORerrorThrown: any) => {
-          if (reqId !== CURRENT_REQ_ID) {
-            return;
-          }
-
-          const xhr = dataOrjqXHR.promise ? dataOrjqXHR : jqXhrORerrorThrown;
-
-          const isSuccess =
-            typeof xhr.status === 'number' &&
-            // Things like DELETE index where the index is not there are OK.
-            ((xhr.status >= 200 && xhr.status < 300) || xhr.status === 404);
-
-          if (isSuccess) {
-            let value = xhr.responseText;
-
-            const warnings = xhr.getResponseHeader('warning');
-            if (warnings) {
-              const deprecationMessages = extractDeprecationMessages(warnings);
-              value = deprecationMessages.join('\n') + '\n' + value;
-            }
-
-            if (isMultiRequest) {
-              value = '# ' + req.method + ' ' + req.url + '\n' + value;
-            }
-
-            results.push({
-              response: {
-                timeMs: Date.now() - startTime,
-                statusCode: xhr.status,
-                statusText: xhr.statusText,
-                contentType: xhr.getResponseHeader('Content-Type'),
-                value,
-              },
-              request: {
-                data: opensearchData,
-                method: opensearchMethod,
-                path: opensearchPath,
-              },
-            });
-
-            // single request terminate via sendNextRequest as well
-            sendNextRequest();
+      try {
+        const httpResponse = await opensearch.send(
+          args.http,
+          opensearchMethod,
+          opensearchPath,
+          opensearchData
+        );
+        if (reqId !== CURRENT_REQ_ID) {
+          return;
+        }
+        const statusCode = httpResponse.response?.status;
+        const isSuccess =
+          statusCode && ((statusCode >= 200 && statusCode < 300) || statusCode === 404);
```

Comment:
minor, should we keep the comments for `404`? // Things like DELETE index where the index is not there are OK.

And do we need check if  httpResponse is undefined -> `httpResponse?.response?.status`